src/platform/target.jl

```diff
@@ -11,23 +11,25 @@ const JLL_ENV_CLANG_TARGETS_MAPPING = Dict(
     "x86_64-apple-darwin14"=>"x86_64-apple-darwin14",
     "x86_64-linux-gnu"=>"x86_64-unknown-linux-gnu",
     "x86_64-linux-musl"=>"x86_64-unknown-linux-musl",
-    "x86_64-unknown-freebsd"=>"x86_64-unknown-freebsd13.2",
+    "x86_64-unknown-freebsd13.2"=>"x86_64-unknown-freebsd13.2",
     "x86_64-w64-mingw32"=>"x86_64-w64-windows-gnu",
 )
 
 triple2target(triple::String) = get(JLL_ENV_CLANG_TARGETS_MAPPING, triple, "unknown")
 
 function __triplet(p::Platform)
     for k in keys(p.tags)
-        if k != "arch" && k != "os" && k != "libc"
+        if k != "arch" && k != "os" && k != "libc" && k != "call_abi" && k != "os_version"
             delete!(p.tags, k)
         end
     end
     t = triplet(p)
-    if os(p) == "macos" && arch(p) == "x86_64"
-        t *= "14"
-    elseif os(p) == "macos" && arch(p) == "aarch64"
-        t *= "20"
+    if os_version(p) === nothing
```

Comment:
I don't know why the user wouldn't specify the version, but here I am just maintaining backward compatibility if they didn't.